src/wp-includes/block-template-utils.php

```diff
@@ -1452,19 +1453,15 @@ function inject_ignored_hooked_blocks_metadata_attributes( $post ) {
 	// We need to build the corresponding `WP_Block_Template` object as context argument for the visitor.
 	// To that end, we need to suppress hooked blocks from getting inserted into the template.
 	add_filter( 'hooked_block_types', '__return_empty_array', 99999, 0 );
-	$template = _build_block_template_result_from_post( $post );
+	$template = _build_block_template_result_from_post( get_post( $post->ID ) );
```

Comment:
Previously templates always had an ID because this ran after they were saved from the database. With this change it won't always have a database entry (e.g. can be loaded from the filesystem) so we need to build the template for both scenarios.